synapse/rest/client/register.py

```diff
@@ -379,6 +380,55 @@ async def on_GET(self, request):
             return 200, {"available": True}
 
 
+class RegistrationTokenValidityRestServlet(RestServlet):
+    """Check the validity of a registration token.
```

Comment:
I've realised that to allow this endpoint to be mounted on a client worker, you simply need to register it here:

https://github.com/matrix-org/synapse/blob/84469bdac773ddb79cfc99f31bbac78d27450682/synapse/app/generic_worker.py#L282-L285

Let's do that (and update the worker docs) afterwards in a separate PR though, in order to land this one quicker. :slightly_smiling_face:

Reply:
Done in bf28876f6764e9ac820ae61cf0f7436b413cb19c